app/controllers/favorites_controller.rb

```diff
@@ -2,16 +2,23 @@
 class FavoritesController < ApplicationController
   def create
     authorize :favorite
-    book = Book.find(params[:book_id])
-    favorite = book.favorites.build
-    favorite.user_id = current_user.id
-    favorite.save
-    render json: favorite, status: :ok
+    current_user.favorite_books << book
```

Comment:
check if book is already marked as favorite, if it is return precondition_failed code